app/assets/javascripts/tagging.js

```diff
@@ -80,3 +86,27 @@ function initTagForm(deletion_path, selector) {
   return el;
 
 }
+
+function geocodeStringAndPan(string, onComplete) {
+  var url = "https://maps.googleapis.com/maps/api/geocode/json?address=" + string.split(" ").join("+");
+  var Blurred = $.ajax({
+      async: false,
+      url: url,
+      complete: function(data) {
+        geometry = data.responseJSON.results[0].geometry.location;
+        lat = geometry.lat;
+        lng = geometry.lng;
+        
+        var geo = [lat, lng];
+
+        if (geo.length > 0) {
+          var r = confirm("This looks like a location. Is this full description of the location accurate?");
+          console.log(geo[0]);
+          console.log(geo[1]);
+          if(r) { 
+            addTag("lat:" + geo[0].toString() + ",lng:" + geo[1].toString()+",place:"+string);
```

Comment:
OK, so I think we can put off refactoring `addTag` for now -- or do it in an independent PR. This should just submit the HTML form. I'm going to merge it!